src/backend/src/integrations/slack.ts

```diff
@@ -9,49 +9,96 @@ const slack = new WebClient(process.env.SLACK_BOT_TOKEN);
  * @param message - the text content of the message being sent
  * @param link - the link for the button on the message
  * @param linkButtonText - the text for the button on the message
+ * @returns the channel id and timestamp of the created slack message
  */
 export const sendMessage = async (slackId: string, message: string, link?: string, linkButtonText?: string) => {
   const { SLACK_BOT_TOKEN } = process.env;
   if (!SLACK_BOT_TOKEN) return;
 
-  // if link and link button are provided, add the button to the message, otherwise just send the markdown block
-  const block =
-    link && linkButtonText
-      ? {
-          type: 'section',
-          text: {
-            type: 'mrkdwn',
-            text: message
-          },
-          accessory: {
-            type: 'button',
-            text: {
-              type: 'plain_text',
-              emoji: true,
-              text: linkButtonText
-            },
-            url: link
-          }
-        }
-      : {
-          type: 'section',
-          text: {
-            type: 'mrkdwn',
-            text: message
-          }
-        };
+  const block = generateSlackTextBlock(message, link, linkButtonText);
 
   try {
-    await slack.chat.postMessage({
+    const response: ChatPostMessageResponse = await slack.chat.postMessage({
       token: SLACK_BOT_TOKEN,
       channel: slackId,
       text: message,
       blocks: [block],
       unfurl_links: false
     });
+
+    return response && response.channel && response.ts ? { channelId: response.channel, ts: response.ts } : null;
   } catch (error) {
     throw new HttpException(500, 'Error sending slack message, reason: ' + (error as any).data.error);
   }
 };
 
+/**
+ * Sends a slack message as a reply in a thread
+ * @param slackId - the channel id of the channel of the message to reply to
+ * @param parentTs - the timestamp of the message to reply to in a thread
+ * @param message - the text content of the message being sent
+ * @param link - the link for the button on the message
+ * @param linkButtonText - the text for the button on the message
+ */
+export const replyToMessageInThread = async (
+  slackId: string,
+  parentTs: string,
+  message: string,
+  link?: string,
+  linkButtonText?: string
+) => {
+  const { SLACK_BOT_TOKEN } = process.env;
+  if (!SLACK_BOT_TOKEN) return;
+
+  const block = generateSlackTextBlock(message, link, linkButtonText);
+
+  try {
+    await slack.chat.postMessage({
+      token: SLACK_BOT_TOKEN,
+      channel: slackId,
+      thread_ts: parentTs,
+      text: message,
+      blocks: [block],
+      unfurl_links: false
```

Comment:
What's this doing?

Reply:
its a separate function for sending slack messages as a reply to a thread since in order for a message to be a reply, it requires a timestamp. if i added it as an optional arg to sendMessage, then it would either have to be before or after the link/linkButtonText params, meaning whichever of the args were the later half would have to coexist with the first arg, but all messages sent as a reply dont necessarily need to have a button, and vice versa. if you wanted to put that logic into sendMessage, then link/linkButtonText and ts would have to be consolidated into an optional options? param, which is an object with all three params.

Reply:
sorry I meant specifically, unfurl_links: false

Reply:
removed